core-common/src/main/java/org/glassfish/jersey/message/internal/MessagingBinders.java

```diff
@@ -137,4 +150,110 @@ protected void configure() {
             return providers;
         }
     }
+
+    private static final class EnabledProvidersBinder {
+        private enum Provider {
+            DOMSOURCE ("javax.xml.transform.dom.DOMSource"),
+            RENDEREDIMAGE ("java.awt.image.RenderedImage"),
+            SAXSOURCE ("javax.xml.transform.sax.SAXSource"),
+            SOURCE ("javax.xml.transform.Source"),
+            STREAMSOURCE ("javax.xml.transform.stream.StreamSource");
+            Provider(String className) {
+                this.className = className;
+            }
+            private String className;
+        }
+
+        private static final String ALL = "ALL";
+        private HashSet<Provider> disabledProviders = new HashSet<>();
```

Comment:
I would implement this logic vice-versa / instead of adding disabled providers to a set I would create the set of all providers and then remove disabled. So, you will have set of enabled providers only. For ALL this would be empty set. In such a case you would eliminate all IF CONTAINS checks and just loop over already prepared set thus you know all elements of that set are enabled.

Reply:
I did not like the logic that removes the disabled providers when the set of enabled would be created. If only one provider would be disabled (such as in the case of JDK11.desktop), the `if contains()` would still need to be there, checking whether the single provider is not disabled.

Reply:
I got your point. I have changed that.